src/main/java/com/pingcap/tikv/PDClient.java

```diff
@@ -98,7 +98,7 @@ public TiRegion getRegionByKey(ByteString key) {
                 .setRegionKey(key)
                 .build();
 
-        PDErrorHandler<GetRegionResponse> handler = new PDErrorHandler<>(r -> r.getHeader().getError());
+        PDErrorHandler<GetRegionResponse> handler = new PDErrorHandler<>(r -> r.getHeader().getError(), r -> r.getHeader().hasError());
```

Comment:
little bit over kill and expose to much detail to caller code. If just for test hasError, that's too much.